client/my-sites/plugins/hooks/types.ts

```diff
@@ -0,0 +1,25 @@
+// NOTE: This is a crude version of an enum,
+// since TypeScript enums behave strangely
+// with regular JavaScript after transpilation
+export const PluginActions = {
+	ACTIVATE: 'activate',
+	DEACTIVATE: 'deactivate',
+	REMOVE: 'remove',
+	UPDATE: 'update',
+	ENABLE_AUTOUPDATES: 'enableAutoUpdates',
+	DISABLE_AUTOUPDATES: 'disableAutoUpdates',
+} as const;
+
+export type PluginActionName = ( typeof PluginActions )[ keyof typeof PluginActions ];
+
+export type Site = {
+	ID: number;
+	title: string;
+	canUpdateFiles?: boolean;
+};
+
+export type Plugin = {
+	slug: string;
+	sites: Record< string, unknown >;
```

Comment:
Just a question because I haven't seen it before.... what is this Record of which you speak?

Reply:
Ah, yeah -- it's one of the built-in TypeScript types, basically an object where (in this case) the keys are `string`s and the values are of some unknown type.